File: order.service/src/main/java/br/com/edu/order/infra/rest/v1/request/ItemRequest.java
package br.com.edu.order.infra.rest.v1.request;

import br.com.edu.order.domain.Item;
import lombok.Data;

import java.math.BigDecimal;

@Data
public class ItemRequest {

    private String product;

    private BigDecimal price;

    private Integer quantity;

    public ItemRequest() {
        product = "";
        price = BigDecimal.ZERO;
        quantity = 0;
    }

    public Item toModelItem() {
        var item = new Item();
        item.setProduct(this.product);
        item.setPrice(this.price);
        item.setQuantity(this.quantity);
        return item;
    }
}

File: order.service/order-service-app/src/main/kotlin/br/com/edu/order/infra/rest/v1/request/ItemRequest.kt
package br.com.edu.order.infra.rest.v1.request

import com.fasterxml.jackson.annotation.JsonIgnoreProperties
import com.fasterxml.jackson.annotation.JsonProperty
import br.com.edu.order.domain.Item
import java.math.BigDecimal

@JsonIgnoreProperties(ignoreUnknown = true)
data class ItemRequest(

    @JsonProperty("product")
    val product: String = "",

    @JsonProperty("price")
    val price: BigDecimal = BigDecimal.ZERO,

    @JsonProperty("quantity")
    val quantity: Int = 0

) {

    fun toModelItem(): Item {
        val item = Item()
        item.product = this.product
        item.price = this.price
        item.quantity = this.quantity
        return item
    }

}
